linux/system/repo.sls

```diff
@@ -91,6 +91,16 @@ linux_repo_{{ name }}_key:
 
 {%- elif repo.key_url|default(False) %}
 
+{%- if system.proxy.get('pkg', {}).get('enabled', False) and system.proxy.get('pkg', {}).get('http', None) %}
+
+linux_repo_{{ name }}_key:
+  cmd.wait:
+    - name: "curl --proxy {{ system.proxy.get('pkg', {}).get('http', None) }}  -s {{ repo.key_url }} | apt-key add -"
```

Comment:
Sounds like some more pull requests are needed to clean up this formula. README and current implementation do not follow this pattern.